File: Stack/StackArray.c
#include "stdio.h"
#include "stdlib.h"

struct Stack
{
    int size;
    int top;
    int *s;
};

void Create(struct Stack* st){
    printf("Enter the size of the stack: ");
    scanf("%d", &st->size);
    st->top = -1;
    st->s = (int *)malloc(st->size * sizeof(int));
    int el;
    for(int i=0;i<st->size;i++){
        printf("Enter element %d: ",i+1);
        scanf("%d",&el);
        st->s[i] = el;
    }
}

void Display(struct Stack st){
    for (int i=st.top;i>=0;i--){
        printf("%d ", st.s[i]);
    }
}

void push(struct Stack *st, int val)
{
    if (st->top == st->size - 1)
        printf("Stack Overflow!");
    else
    {
        st->top++;
        st->s[st->top] = val;
    }
}

int pop(struct Stack *st){
    int el = -1;
    if (st->top==-1)
        printf("Stack Underflow!");
    else{
        el = st->s[st->top];
        st->top--;
    }
    return el;
}

int peak(struct Stack st, int pos){
    int x=-1;
    if (st.top<pos)
        printf("Invalid Position\n");
    else{
        x = st.s[st.top-pos];
    }
    return x;
}

int isFull(struct Stack st){
    if(st.top+1==st.size)
        return 1;
    else
        return 0;
}

int isEmpty(struct Stack st){
    if (st.top==-1)
        return 1;
    else
        return 0;
}

int top(struct Stack st) {
    if (st.top==-1)
        return -1;
    else
        return st.s[st.top];
}

int main()
{
    struct Stack st;
    Create(&st);
    Display(st);
}

File: Stack/StackArray.py

class Stack:
    def __init__(self):
        self.size = 0
        self.top = -1
        self.s = []
        
    def Create(self):
        self.size = int(input("Enter size of the Stack: "))
        self.top = -1
        self.s = [0] * self.size
    
    def Display(self):
        for i in range(self.top,-1,-1):
            print(f"{self.s[i]}", end=" ")
    
    def isEmpty(self):
        return self.top==-1
    
    def isFull(self):
        return self.top+1==self.size
    
    def stackTop(self):
        if self.isEmpty(): return -1
        else: return self.s[self.top]
    
    def Push(self, val):
        if self.isFull():
            print("Stack Overflow!")
        else:
            self.top += 1
            self.s[self.top] = val
    
    def Pop(self):
        x = -1
        if self.isEmpty():
            print("Stack Underflow!")
        else:
            x = self.s[self.top]
            self.top -= 1
        return x
    
    def Peak(self, pos):
        x = -1
        if pos>self.top:
            print("Invalid positon")
        else:
            x = self.s[self.top-pos]
        return x
    
st = Stack()
st.Create()
print(st.isEmpty())
st.Push(1)
print(st.isEmpty())
st.Push(2)
print(st.Peak(0))
print(st.isFull())
st.Push(3)
print(st.Pop())
st.Push(4)
st.Push(5)
print(st.stackTop())
print(st.isFull())
st.Display()